sympy/solvers/solveset.py

```diff
@@ -3888,6 +3888,21 @@ def _handle_poly(polys, symbols):
 
     return poly_sol, poly_eqs
 
+def expand_and_simplify(subs_res):
```

Comment:
`simplify` should not be used (in general) to fix very specific problems. As pointed out in the issue, the problem stems from how 2-arg Adds are handled. Apparently, two different code paths produced the same solution but did so in a different way and one leaves a negative one factored out while the other has it distributed. One solution is to just patch things up here with a routine like
```python
def touchup(sol):
    # put the minimal simplification needed to keep duplicate
    # solutions from appearing in the solution
    return bottom_up(sol, signsimp)
```
(and import bottom_up on the line importing from traversal).

A better solution would be to track down why these two expressions are being constructed in two different ways and see if they should be doing things the same way or using common code.

The best solution would be to disable distribution of the constant onto an Add (but that is a very involved task).